Marlin/ultralcd.cpp

```diff
@@ -924,6 +945,11 @@ void kill_screen(const char* lcd_msg) {
       return mesh_edit_value;
     }
 
+    float lcd_manual_probe() {
+      lcd_goto_screen(_lcd_mesh_edit_NOP);
```

Comment:
How does other code manage interaction with the LCD without requiring `_lcd_mesh_edit_NOP` and other hacks? Take a look at the methods used by `gcode_M600` to take over the screen for display and feedback purposes. Take a look at how `LCD_BED_LEVELING` does its UI as an LCD-initiated process.

For U.B.L. I'd recommend following the example of `gcode_M600`. In the process of fixing the LCD interaction code, I'm sure new opportunities for further unification and improvement of our UI methods will jump out.

The main thing is, when we see we're using hacks, it's time to take a step back.